SignalMessaging/utils/ConversationSearcher.swift

```diff
@@ -1,19 +1,77 @@
 //
-//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
 //
 
 import Foundation
 import SignalServiceKit
 
+public class SearchResult {
+    public let thread: ThreadViewModel
+    public let snippet: String?
```

Comment:
We need to be careful about assuming we can represent contact matches with `ThreadViewModel`.  Not all contacts have threads.  We want to avoid creating an unsaved `TSThread` instance and/or overwriting `TSThread` instances.

Reply:
Right - I'm assuming we shouldn't represent contact matches with a threadViewModel and that this will change somehow.